Lib/test/test_pkgutil.py

```diff
@@ -522,7 +522,44 @@ def test_mixed_namespace(self):
         del sys.modules['foo.bar']
         del sys.modules['foo.baz']
 
-    # XXX: test .pkg files
+
+    def test_extend_path_argument_types(self):
+        pkgname = 'foo'
+        dirname_0 = self.create_init(pkgname)
+
+        # If the input path is not a list it is returned unchanged
+        self.assertEqual('notalist', pkgutil.extend_path('notalist', 'foo'))
+        self.assertEqual(('not', 'a', 'list'), pkgutil.extend_path(('not', 'a', 'list'), 'foo'))
+        self.assertEqual(123, pkgutil.extend_path(123, 'foo'))
+        self.assertEqual(None, pkgutil.extend_path(None, 'foo'))
+
+        # Cleanup
+        shutil.rmtree(dirname_0)
+        del sys.path[0]
+
+
+    def test_extend_path_pkg_files(self):
+        pkgname = 'foo'
+        dirname_0 = self.create_init(pkgname)
+
+        with open(os.path.join(dirname_0, 'bar.pkg'), 'w') as pkg_file:
+            pkg_file.write('\n'.join([
+                'baz',
+                '/foo/bar/baz',
+                '',
+                '#comment'
+            ]))
+
+        # A *.pkg file is trusted at face value. Apart from checking for duplicates, all entries found in a *.pkg
+        # file are added as-is to the resulting paths, regardless of whether they exist on the filesystem.
+        extended_paths = pkgutil.extend_path(sys.path, 'bar')
+
+        self.assertEqual(extended_paths[-2], 'baz')
+        self.assertEqual(extended_paths[-1], '/foo/bar/baz')
```

Comment:
Please also
- check that `extended_paths[:-2]` is the same as the original path
- test that duplicates are removed.


The documentation doesn't make it clear that comments are ignored. Could you note that in the docs?

Perhaps change:

    apart from checking for duplicates, all entries ...

to:

    blank lines, comments and duplicates are ignored; all other entries ...

Reply:
Turns out that `extend_path` does happily insert duplicates in the list. To not introduce a potentially breaking change, I've adjusted the documentation accordingly. Also, the suggested check that `extended_paths[:-2]` equals the original path has been added.